src/charm.py

```diff
@@ -1285,7 +1288,9 @@ def server_cert_sans_dns(self) -> List[str]:
             name, _, _ = socket.gethostbyaddr(target)  # type: ignore
             # Do not return "hostname" like '10-43-8-149.kubernetes.default.svc.cluster.local'
             if is_hostname(name) and not name.endswith(".svc.cluster.local"):
-                return [name]
+                # In case we can do a DNS lookup on that IP address,
+                #  return the resolved hostname as well as the IP address to be both included in the certificate SANS.
+                return [name, target] if target else [name]
```

Comment:
I think it is rather strange that we are doing a reverse DNS lookup here. I think if hostname is not set we should just use the IP address. In other words the cert should match the address in the "Serving at \<address\>" status message.

So I think the best course of action is to just delete this block altogether.

Reply:
On second thought, it is probably just safer to keep this.